pkg/auth/agent_local_authenticator.go

```diff
@@ -0,0 +1,114 @@
+package auth
+
+import (
+	"crypto"
+	"encoding/base64"
+	"net/http"
+	"os"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/security"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/gencrypto"
+	"github.com/openshift/assisted-service/pkg/ocm"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type AgentLocalAuthenticator struct {
+	log       logrus.FieldLogger
+	publicKey crypto.PublicKey
+}
+
+func NewAgentLocalAuthenticator(cfg *Config, log logrus.FieldLogger) (*AgentLocalAuthenticator, error) {
+	if cfg.ECPublicKeyPEM == "" {
+		return nil, errors.Errorf("agent installer local authentication requires an ecdsa Public Key")
+	}
+
+	// When generating an agent ISO, the Agent installer creates ECDSA public/private keys.
+	// However, the systemd services of the Agent installer fail to parse multiline keys accurately.
+	// To address this, the keys are encoded in base64 format to condense them into a single line
+	// before being transmitted to the assisted service.
+	// Upon reception, the assisted service decodes these keys back to their original multiline format
+	// for subsequent processing.
+
+	decodedECPublicKeyPEM, err := base64.StdEncoding.DecodeString(cfg.ECPublicKeyPEM)
+	if err != nil {
+		log.WithError(err).Fatal("Error decoding public key:")
+	}
+	cfg.ECPublicKeyPEM = string(decodedECPublicKeyPEM)
+
+	decodedECPrivateKeyPEM, err := base64.StdEncoding.DecodeString(cfg.ECPrivateKeyPEM)
+	if err != nil {
+		log.WithError(err).Fatal("Error decoding private key:")
+	}
+	cfg.ECPrivateKeyPEM = string(decodedECPrivateKeyPEM)
+	os.Setenv("EC_PRIVATE_KEY_PEM", string(decodedECPrivateKeyPEM))
```

Comment:
The private key is used here because it is needed in the existing code here https://github.com/openshift/assisted-service/blob/478cd74e4b5c32bc237b2a132c0ce1c3e124de90/internal/gencrypto/token.go#L25-L30 and `LocalJWT` is invoked from https://github.com/pawanpinjarkar/assisted-service/blob/ff6deee932e48a792156077e134586fc3246ce3f/internal/cluster/auth.go#L33-L34

Reply:
Use of private keys is not required after updating the case statements in the relevant code flow as above.